File: app/src/main/java/pw/janyo/whatanime/interfaces/SearchService.java
package pw.janyo.whatanime.interfaces;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface SearchService {
    @FormUrlEncoded
    @POST("/api/search")
	Observable<ResponseBody> search(@Query("token") String token, @Field("image") String image, @Field("filter") String filter);
}

File: app/src/main/java/pw/janyo/whatanime/interfaces/SearchService.kt
package pw.janyo.whatanime.interfaces

import pw.janyo.whatanime.classes.Animation
import retrofit2.Call
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface SearchService {
    @FormUrlEncoded
    @POST("/")
    fun search(@Field("image") image: String, @Field("filter") filter: String?): Call<Animation>
}